### DISCUSSION TITLE
How to remove DC offset from AnalogAudioStream

### DISCUSSION THREAD

**Discussion by ElBartoME** [pschatzmann/arduino-audio-tools#930] [Q&A] [answered]
Hi,

I'm trying to use the library to read a microphone and then use the FFT function to generate the frequency spectrum. I'm having problems because the microphone gives out a 1.65V reference and therefore all values are shifted upwards.

I'm not sure how you did it in your example but how do you remove the DC with the library? I was not able to figure that out.

Kind regards

**Comment by pschatzmann** [comment, marked as answer]
Originally I have used (and provided) the ConverterAutoCenter to do this. Alternatively you can also use the ConverterScaler.
Recently I have introduced the is_auto_center_read configuration parameter which will do this converstion automatically. By default this is set to active.
Today I committed the change to activate this functionality on the ESP32. 

So, with the latest release no conversion should be necessary any more.

**Reply by ElBartoME** [reply]
Thank you very much! I knew about the is_auto_center but it didn't have any effect on the ESP32. Thanks for the update. It works great now.
